src/server/services/afis/afis.ts

```diff
@@ -0,0 +1,170 @@
+      grant_type: 'client_credentials',
+      client_secret: process.env.BFF_AFIS_OAUTH_CLIENT_SECRET,
+    }),
+    transformResponse: (responseData: OathResponseData) =>
+      `${responseData.token_type} ${responseData.access_token}`,
+  };
+
+  const dataRequestConfig = getApiConfig('AFIS_OAUTH', baseConfig);
+
+  const response = await requestData<string>(
+    dataRequestConfig,
+    requestID,
+    authProfileAndToken
+  );
+
+  return response;
+}
+
+export type JaOfNee = 'Ja' | 'Nee';
+
+/** Business partner private response from external AFIS API.
+ *
+ *  # Properties
+ *
+ *  Record.BSN - Is a string when there is a leading zero present, otherwise a number.
+ */
+export type AFISBusinessPartnerPrivateSourceResponse = {
+  BSN: number | string;
+  Gevonden: JaOfNee;
+  Zakenpartnernummer?: string;
+  Blokkade?: JaOfNee;
+  Afnemers_indicatie?: JaOfNee;
+};
+
+/** Business partner commercial response from external AFIS API.
+ *
+ *  # Properties
+ *
+ *  Record.KVK - Is a string when there is a leading zero present, otherwise a number.
+ */
+type AFISBusinessPartnerRecord = {
+  KVK: number | string;
+  Zakenpartnernummer: string;
+  Vestigingsnummer?: string;
+  Blokkade: JaOfNee;
+  Gevonden: JaOfNee;
+};
+
+export type AFISBusinessPartnerCommercialSourceResponse = {
+  Record: AFISBusinessPartnerRecord | AFISBusinessPartnerRecord[];
+};
+
+export type BusinessPartnerKnownResponse = { isKnown: boolean };
+
+/** Returns if the person logging in is known in the AFIS source API */
+export async function fetchIsKnownInAFIS(
+  requestID: requestID,
+  authProfileAndToken: AuthProfileAndToken
+) {
+  const bearerTokenResponse = await fetchAFISBearerToken(
+    requestID,
+    authProfileAndToken
+  );
+  if (bearerTokenResponse.status === 'ERROR') {
+    return bearerTokenResponse;
+  }
+
+  let profileType: 'BSN' | 'KVK' = 'BSN';
+  if (authProfileAndToken.profile.profileType === 'commercial') {
+    profileType = 'KVK';
+  }
+
+  const baseConfig: DataRequestConfig = {
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: bearerTokenResponse.content,
+    },
+    data: {
+      [profileType]: authProfileAndToken.profile.id,
+    },
+    transformResponse: transformBusinessPartnerisKnown,
+  };
+
+  const dataRequestConfig = getApiConfig('AFIS_BUSINESSPARTNER', baseConfig);
+
+  dataRequestConfig.url = appendBusinessPartnerEndpoint(
+    dataRequestConfig.url,
+    authProfileAndToken.profile
+  );
+
+  const response = await requestData<BusinessPartnerKnownResponse>(
+    dataRequestConfig,
+    requestID,
+    authProfileAndToken
+  );
+
+  return response;
+}
+
+function appendBusinessPartnerEndpoint(
+  url: string | undefined,
+  authProfile: AuthProfile
+) {
+  // RP Note: Is there a better place to do a check like this?
+  if (!url) {
+    throw new Error(
+      `url is ${url}, please define this in the ApiConfig (src/server/config.ts)`
+    );
+  }
```

Comment:
Dit kun je verwijderen, zeker in dergelijke weinig gebruikte (en niet geëxporteerde) functies is dergelijke error afhandeling niet nodig. 

Je kunt dit overigens ook met TS afhandelen dor `url: string` te doen en de `| undefined` weg te halen.